cmd/tracee-ebpf/main.go

```diff
@@ -209,54 +217,56 @@ func main() {
 			&cli.StringSliceFlag{
 				Name:    "trace",
 				Aliases: []string{"t"},
-				Value:   nil,
+				Value:   cli.NewStringSlice(config.Config.Trace...),
 				Usage:   "select events to trace by defining trace expressions. run '--trace help' for more info.",
 			},
 			&cli.StringSliceFlag{
 				Name:    "capture",
 				Aliases: []string{"c"},
-				Value:   nil,
+				Value:   cli.NewStringSlice(config.Config.Capture...),
 				Usage:   "capture artifacts that were written, executed or found to be suspicious. run '--capture help' for more info.",
 			},
 			&cli.StringSliceFlag{
 				Name:    "output",
 				Aliases: []string{"o"},
-				Value:   cli.NewStringSlice("format:table"),
+				Value:   cli.NewStringSlice(append([]string{"format:table"}, config.Config.Output...)...),
 				Usage:   "Control how and where output is printed. run '--output help' for more info.",
 			},
 			&cli.IntFlag{
 				Name:    "perf-buffer-size",
 				Aliases: []string{"b"},
-				Value:   1024,
+				Value:   config.Config.DefaultEventSubmitBufferSize(),
 				Usage:   "size, in pages, of the internal perf ring buffer used to submit events from the kernel",
 			},
 			&cli.IntFlag{
 				Name:  "blob-perf-buffer-size",
-				Value: 1024,
+				Value: config.Config.DefaultBlobBufferSize(),
 				Usage: "size, in pages, of the internal perf ring buffer used to send blobs from the kernel",
 			},
 			&cli.BoolFlag{
 				Name:        "debug",
-				Value:       false,
+				Value:       config.Config.VerboseDebug,
 				Usage:       "write verbose debug messages to standard output and retain intermediate artifacts",
 				Destination: &debug,
 			},
 			&cli.StringFlag{
 				Name:        "install-path",
-				Value:       "/tmp/tracee",
+				Value:       config.Config.DefaultInstallPath(),
 				Usage:       "path where tracee will install or lookup it's resources",
 				Destination: &traceeInstallPath,
 			},
 			&cli.StringFlag{
 				Name:        "build-policy",
-				Value:       "if-needed",
+				Value:       config.Config.DefaultBuildPolicy(),
 				Usage:       "when to build the bpf program. possible options: 'never'/'always'/'if-needed'",
 				Destination: &buildPolicy,
 			},
 		},
 	}
 
-	err := app.Run(os.Args)
+	app.Before = altsrc.InitInputSourceWithContext(app.Flags, altsrc.NewJSONSourceFromFlagFunc("config"))
```

Comment:
I'm not entirely clear on why we need altsrc now if you're parsing the JSON manually

Reply:
That's a leftover, i'll clean it up.

Reply:
Done